src/custom/components/MenuDropdown/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react'
+import { MenuFlyout, Content } from './styled'
+import IMAGE_CARRET_DOWN from 'assets/cow-swap/carret-down.svg'
+import SVG from 'react-inlinesvg'
+import { useMediaQuery, upToLarge } from 'hooks/useMediaQuery'
+
+interface MenuProps {
+  title: string
+  children: React.ReactNode
+}
+
+export function Menu({ title, children }: MenuProps) {
+  const isUpToLarge = useMediaQuery(upToLarge)
+  const [showMenu, setShowMenu] = useState(false)
+  const handleOnClick = () => isUpToLarge && setShowMenu(!showMenu)
```

Comment:
SetStates allow passing a callback as a param so it's safer to do:
```ts
setShowMenu(menuState => !menuState)
```

Reply:
Like this ?
`const handleOnClick = () => isUpToLarge && setShowMenu((showMenu) => !showMenu)`

Why is it safer?